### ISSUE TITLE
[FEATURE]: Custom loss functions

### ISSUE THREAD

**Issue by dphogan** [CosmiQ/solaris#308]
**Is your feature request related to a problem? Please describe.**
Currently, Solaris can only be used with built-in loss-functions.

**Describe the solution you'd like**
Ideally, users could include the names of custom loss functions in their YAML file and then feed in the classes/functions when calling Trainer(), just as they currently can do for custom models.

**Describe alternatives you've considered**
An alternative is including the name and module for custom loss functions directly in the YAML file, but that is not as convenient.

**Comment by dphogan** [comment]
Resolved with pull request #309
